### ISSUE TITLE
When create vmware vsphere nodetemplate UI sends some fields as ints

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#19032]
Version: Master 2.2 (3/19/19) ImageID: 1e8bf89a496f

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Create a vmware vsphere node template with defaults and watch HTTP
- UI seems to send cpuCount, diskSize, memorySize, vcenterPort as ints but these are of type string in schema

**Result:**
This actually doesn't seem to hurt anything but was basically a "warning flag" for QA. We saw that when we inspect the node template in the API it's correctly entered as strings even though request payload from client-side did not include as strings.

Client HTTP Request Payload (POST) nodetemplate
```
cloudinit: ""
cpuCount: 2
datacenter: ""
datastore: ""
diskSize: 20000
folder: ""
hostsystem: ""
memorySize: 2048
network: []
pool: ""
type: "vmwarevsphereConfig"
username: ""
vappIpallocationpolicy: ""
vappIpprotocol: ""
vappProperty: []
vappTransport: ""
vcenter: ""
vcenterPort: 443
```

View node template in API:
```
"cloudinit": "",
"cpuCount": "2",
"datacenter": "",
"datastore": "",
"diskSize": "20000",
"folder": "",
"hostsystem": "",
"memorySize": "2048",
"network": null,
"pool": "",
"username": "",
"vappIpallocationpolicy": "",
"vappIpprotocol": "",
"vappProperty": null,
"vappTransport": "",
"vcenter": "",
"vcenterPort": "443"
```

**Other details that may be helpful:**
_In v2.1.7 the UI sends cpuCount and vcenterPort as string but diskSize and memorySize are ints._

**Comment by loganhz** [comment]
Hi @davidnuzik 

Both 2.2 and 2.1 are doing the same thing now.

1. UI sends the `int` to backend.
2. Backend converts `int` to `string` and save them.
3. View API shows `string`

So they have the same behavior. We don't need to worry about it in 2.2.

To fix it, we need to change https://github.com/rancher/ui/blob/master/lib/nodes/addon/components/driver-vmwarevsphere/component.js#L151-L155 to string.

But converting `int` to `string` in this case is kind of a feature of rancher server. So we don't need to change it.

Move it out from 2.2

**Comment by davidnuzik** [comment]
Thanks @loganhz sounds good; we were just concerned is all. Glad there is nothing to worry about here.

**Comment by deniseschannon** [comment]
@cjellick @luthermonson with the changes of vsphere, can you follow up with the UI team on if anything needs to be changed in this area based on the API

**Comment by codyrancher** [comment]
I made the change relevant to this as part of this PR https://github.com/rancher/ui/pull/3449

**Comment by westlywright** [comment]
Available in `latest-2.3`.

**Comment by bmdepesa** [comment]
rancher/rancher:v2.3-head 6611960c9
UI: `latest-2.3`

* Created vSphere node template
* All fields are being sent as strings:
```
...
cpuCount: "2"
diskSize: "20000"
memorySize: "2048"
sshPort: "22"
vcenterPort: "443"
...
```
